izettle-messaging/src/main/java/com/izettle/messaging/MessagePublisher.java

```
@@ -1,8 +1,12 @@
 package com.izettle.messaging;
 
 import java.util.Collection;
+import java.util.Map;
 
 public interface MessagePublisher {
     <M> void post(M message, String eventName) throws MessagingException;
     <M> void postBatch(Collection<M> messages, String eventName) throws MessagingException;
+    default <M> void post(M message, String eventName, Map<String, String> attributes) throws MessagingException {
```

We talked about using a default method - but the rational behind hiding it in a sub-interface was that we don't want to affect existing code - mainly tests.

---

The existing code would not use this new method?

---

Ah you are correct

---

Thanks, :)